### PR TITLE
Remove oauth2client._helpers dependency

### PR DESCRIPTION
**Pull request by helen-fornazier**
oauth2client is deprecated
Migrate required helper's code from oauth2client to googleapiclient to
decrease our dependency on oauth2client

_helpers.py copied from oauth2client code
https://github.com/google/oauth2client/blob/master/oauth2client/_helpers.py

### CODE REVIEW

**Comment by helen-fornazier** [comment]
Please review, I am not entirely sure if this is the appropriated solution.
But with this change and #492 I can already manipulate instances using just google-auth, after uninstalling oauth2client

More specifically, what I am trying to run with google-auth instead of oauth2client is:
[code]: https://github.com/ClusterLabs/fence-agents/blob/master/fence/agents/gce/fence_gce.py
[PR]: https://github.com/ClusterLabs/fence-agents/pull/197

Thanks

**Comment by helen-fornazier** [comment]
@jonparrott done, I imported the tests too as suggested. Could you please take a look in the commits?

Thanks

**Comment by helen-fornazier** [comment]
Also, now the only file inside `googleapiclient/` who uses oauth2client is `googleapiclient/sample_tools.py`, do you know if this module is used by external libraries?
It seems to me that this file is just an utility module used by `samples/`, so I was wondering if we could move `sample_tools.py`to the `samples/` folder and completely remove oauth2client from the `install_requires` list in the setup.py

**Comment by helen-fornazier** [comment]
@jonparrott lgtm, I added your suggestion in the PR, could you take a look please?

Can I remove oauth2client from the install_requires list or replace it by google-auth? https://github.com/google/google-api-python-client/blob/master/setup.py#L67

File: googleapiclient/sample_tools.py [hunk 1]
```diff
@@ -27,9 +27,13 @@
...
 from googleapiclient.http import build_http
-from oauth2client import client
-from oauth2client import file
-from oauth2client import tools
+
+try:
+    from oauth2client import client
+    from oauth2client import file
+    from oauth2client import tools
+except:
```
Comment (jonparrott) [inline googleapiclient/sample_tools.py]: Avoid broad excepts, use `except ImportError`.

**Comment by jonparrott** [comment]
> Can I remove oauth2client from the install_requires list or replace it by google-auth? 

Yes, but let's two in two releases: one to have this, and then a second one to remove oauth2client.

File: googleapiclient/sample_tools.py (hunk 1 above)
Comment (helen-fornazier) [reply googleapiclient/sample_tools.py]: Fixed

**Comment by jonparrott** [comment]
@helen-fornazier thank you so much for doing this!

**Pull request merged**